common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -48,6 +59,28 @@ public Job createJob(String resourceTypes, String url) {
         job.setProgress(0);
         job.setUser(userService.getCurrentUser());
 
+        if (contractNumber != null) {
+            contractRepository.findContractByContractNumber(contractNumber).ifPresentOrElse(contractFound -> {
+                User user = userService.getCurrentUser();
+                Sponsor userSponsor = user.getSponsor();
+                Sponsor contractSponsor = contractFound.getSponsor();
+                if (!userSponsor.getHpmsId().equals(contractSponsor.getHpmsId())) {
+                    log.error("User {} does not have permissions for contract {}", user.getUsername(), contractFound.getContractNumber());
+                    throw new UserPermissionsException("User " + user.getUsername() + " does not have permissions for contract " + contractFound.getContractNumber());
```

Comment:
`Username` could be an email address, which is PII: let's suppressed it from the exception message.